src/Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource/ActivitySourceInput.cs

```diff
@@ -0,0 +1,289 @@
+            e.Payload["Name"] = activity.DisplayName;
+            e.Payload["SpanId"] = activity.Id;
+            e.Payload["ParentSpanId"] = activity.ParentSpanId.ToHexString();
+            e.Payload["StartTime"] = e.Timestamp;
+            if (activity.Duration != TimeSpan.Zero)
+            {
+                e.Payload["EndTime"] = activity.StartTimeUtc + activity.Duration;
+            }
+            e.Payload["TraceId"] = activity.TraceId.ToHexString();
+            if (ActivityKindNames.TryGetValue(activity.Kind, out string activityKindName))
+            {
+                e.Payload["SpanKind"] = activityKindName;
+            }
+            e.Payload["IsRecording"] = activity.Recorded;
+
+            // The following property additions may cause name conflicts, so using AddPayloadProperty() to handle them.
+            foreach(var el in activity.Baggage)
+            {
+                AddPayloadProperty(e, el.Key, el.Value);
+            }
+
+            AddPayloadProperty(e, "ActivitySourceName", activity.Source.Name);
+
+            if (capturedData == ActivitySamplingResult.AllData || capturedData == ActivitySamplingResult.AllDataAndRecorded)
+            {
+                // Activity.Tags is a subset of Activity.TagObjects that have string value,
+                // so it is sufficient to just iterate over Activity.TagObjects to capture all activity tags.
+                foreach(var tagObject in activity.TagObjects)
+                {
+                    AddPayloadProperty(e, tagObject.Key, tagObject.Value);
+                }
+
+                if (activity.Events.Any())
+                {
+                    AddPayloadProperty(e, "Events", activity.Events);
+                }
+
+                if (activity.Links.Any())
+                {
+                    AddPayloadProperty(e, "Links", activity.Links);
+                }
+            }
+
+            return e;
+        }
+
+        private void AddPayloadProperty(EventData e, string propertyName, object propertyValue)
+        {
+            e.AddPayloadProperty(propertyName, propertyValue, healthReporter_, nameof(ActivitySourceInput));
+        }
+
+        private void FlushSamplingInfoCacheIfNeeded()
+        {
+            if (activitySampling_.Count > SamplingDecisionCacheFlushThreshold)
+            {
+                lock(activitySampling_)
+                {
+                    if (activitySampling_.Count > SamplingDecisionCacheFlushThreshold)
+                    {
+                        activitySampling_.Clear();
+                    }
+                }
+            }
+        }
+
+        private void WarnAboutIneffectiveEntries(ActivitySourceInputConfiguration configuration_, IHealthReporter healthReporter)
+        {
+            // If (SomeSource, *) configuration entry is succeeded by (SomeSource, SomeActivity) configuration entry,
```

Comment:
This is probably nitpick but theoretically user can specify something like ```("", ActivityName)```, should we do something regarding this?

Reply:
Yes, `(*, ActivityName)` is supported, and we have tests checking that this kind of configuration works.